### DISCUSSION TITLE
Can't search repository for months. "Not indexed yet" "Try again later"

### DISCUSSION THREAD

**Discussion by LeoDavidson** [community/community#107482] [Other Feature Feedback, Questions, & Ideas]
I'm unable to search my own small repository for strings in code.

The error message says the repository has not been indexed yet, and to try again later.

But I've been trying for months now, I think, and the indexing never happens.

At the moment, I have to download a zip of my repository in order to search it locally.

Full error message, which includes the repository name:

```
This repository's code has not been indexed yet. Try again later.
repo:LeoDavidson/discourse-directoryopus Directory
```

**Comment by jelmerk** [comment]
same

**Comment by mwa-corey** [comment]
I also have been experiencing this for months.
It appears it is only returning the message for certain search terms.

**Comment by diachun** [comment]
I am having the same issue. Any search inside my repos returns the message "This account's code is being indexed right now. Try again in a few minutes."
Is this feature actually in production for all accounts?

**Reply by sermapu** [reply]
I've solved it. In the GitHub search bar, type “repo:yourUser/yourRepository import”.

**Comment by ricosaurus** [comment]
Same.  I looked for a solution, and supposedly, once an initial search was attempted, the indexing would begin (and one might assume, end).  But that doesn't appear to be the case -- repositories that got the error message a month ago still do today.  It would be more helpful to just be honest about it -- "Github isn't going to index your inconsequential project so quit trying to search it."

**Reply by sermapu** [reply]
I've solved it. In the GitHub search bar, type “repo:yourUser/yourRepository import”.

**Comment by AtomicRaven** [comment]
We pay for Github Enterprise - this is bullshit.  Happens in half our projects.

**Reply by Markeli** [reply]
And how have you fixed it?

**Reply by thesunlover** [reply]
> We pay for Github Enterprise - this is bullshit. Happens in half our projects.

Please let us know how you have fixed that?

**Reply by sermapu** [reply]
I've solved it. In the GitHub search bar, type “repo:yourUser/yourRepository import”.

**Comment by raisch** [comment]
Seriously considering saving my cash and going to a competitor that actually provides the service I pay for.

**Comment by AtomicRobotMan0101** [comment]
This is 100% bullshit.

The **enshitification begins**. 

Who would care to wager there will soon be an option for _"premium indexing"_?

**Comment by devafortun** [comment]
hi, I suggest like this to solve that problem.
1. Confirm Repository Visibility
  Ensure that your repository is public. Private repositories may have limited search capabilities. You can check this by going to your repository settings:
    Navigate to Settings > General > Danger Zone.
    If it's private and you want it to be public, consider changing it (note that this will make your code visible to everyone).
2. Check for Recent Activity:
  GitHub typically indexes repositories based on activity. If your repository has been inactive for a long time, it might not be prioritized for indexing.
  Make a small change (like updating the README or adding a new file) and commit it. This can sometimes trigger re-indexing.
3. Add a README or Other Documentation:
  If your repository lacks a README file or other documentation, add one. GitHub may prioritize indexing repositories that have documentation, as it improves usability.
4. Use the Code Search Beta:
  GitHub has been rolling out new features for code search. If you haven't already, consider opting into the new code search experience:
    Go to your GitHub account settings and look for options related to "Beta features" or "Code Search."
    Enabling this feature may improve search functionality.
5. Wait for Indexing:
  After making changes or triggering actions that should prompt indexing, wait a bit (usually around 10-30 minutes) and try searching again.

If issue persists after trying above steps, I think it is a good idea to contact Github Support!
Best wishes

**Reply by thesunlover** [reply]
I'm currently using the Fine-grained personal access tokens (Preview) is that what you are referring to in point 4 (Use the Code Search Beta)?
Also, I am trying to do my search on a private personal test repo and it is not indexed, until I get access with a PAT to a private corporate repo. Is there a difference in the indexing priorities?

**Reply by harisdev-netizen** [reply]
> hi, I suggest like this to solve that problem.
> 
> 1. Confirm Repository Visibility
>    Ensure that your repository is public. Private repositories may have limited search capabilities. You can check this by going to your repository settings:
>    Navigate to Settings > General > Danger Zone.
>    If it's private and you want it to be public, consider changing it (note that this will make your code visible to everyone).
> 2. Check for Recent Activity:
>    GitHub typically indexes repositories based on activity. If your repository has been inactive for a long time, it might not be prioritized for indexing.
>    Make a small change (like updating the README or adding a new file) and commit it. This can sometimes trigger re-indexing.
> 3. Add a README or Other Documentation:
>    If your repository lacks a README file or other documentation, add one. GitHub may prioritize indexing repositories that have documentation, as it improves usability.
> 4. Use the Code Search Beta:
>    GitHub has been rolling out new features for code search. If you haven't already, consider opting into the new code search experience:
>    Go to your GitHub account settings and look for options related to "Beta features" or "Code Search."
>    Enabling this feature may improve search functionality.
> 5. Wait for Indexing:
>    After making changes or triggering actions that should prompt indexing, wait a bit (usually around 10-30 minutes) and try searching again.
> 
> If issue persists after trying above steps, I think it is a good idea to contact Github Support! Best wishes

My repository is currently active, and I am even working on it currently and still I am facing the following error:

```
This repository's code has not been indexed yet. Try again later.
repo:harisdev-netizen/runaway-theme language:SCSS
```

**Reply by devafortun** [reply]
Yes, there is.
Fine-grained PATs provide more specific permissions and control over repository access compared to the classic PATs, which is crucial for using features like Code Search effectively.
Regarding your search on a private personal test repository, it is important to note that indexing priorities can differ between private personal repositories and private corporate repositories. Typically, private repositories may not be indexed until specific access permissions are granted through a PAT.
In summary, if you want to ensure that your private personal test repository is indexed, you may need to check the permissions associated with your fine-grained PAT and confirm that it has been granted access to that specific repository.
If you allow me to use your Upwork account, I will give you extra income
or if you introduce me someone, I will appreciate with you

**Reply by acampove** [reply]
This is why I have two remotes, one in gitlab and the other in github. I have never had this problem in gitlab, there, this search just works.

**Reply by eyal9** [reply]
This works, made the repository public

**Comment by server-mm404** [comment]
> I'm unable to search my own small repository for strings in code.
> 
> The error message says the repository has not been indexed yet, and to try again later.
> 
> But I've been trying for months now, I think, and the indexing never happens.
> 
> At the moment, I have to download a zip of my repository in order to search it locally.
> 
> Full error message, which includes the repository name:
> 
> ```
> This repository's code has not been indexed yet. Try again later.
> repo:LeoDavidson/discourse-directoryopus Directory
> ```

**Comment by markus-zhang** [comment]
Doesn't work for me either. I guess I need to get another remote.

**Comment by LegalizeAdulthood** [comment]
I have also been experiencing this problem for months; the age of recent changes on the repository seems to have no bearing on whether or not the repository is searchable.

**Comment by gmerton** [comment]
Same here.  Disappointing that there are limitations on private repos.  This is not rocket science.

**Comment by powtac** [comment]
According to https://help.openai.com/en/articles/11145903-connecting-github-to-chatgpt search for the full repo name and add `import` should trigger the indexing.

> You can manually trigger indexing by going to GitHub and performing a search for the repository using this format: `repo:{username/repo_name} import` (e.g., repo:openai/codex import). Note that the index may take ~5-10 minutes before it becomes available to the connector.

**Reply by powtac** [reply]
Maybe it is to add type=import instead code.

**Comment by unowarder01** [comment]
Same issue

**Reply by unowarder01** [reply]
I have just added `README.md`  and wow my repo is indexed now.......
<img width="460" height="245" alt="image" src="https://github.com/user-attachments/assets/2d12c6ca-0bc0-4cf7-b967-0c70c81c4851" />

**Comment by LegalizeAdulthood** [comment]
It does seem to be working again for me...

**Comment by tanvishinde017** [comment]
Hi @LeoDavidson,

If the repository has remained unindexed for months, that definitely sounds longer than expected for a small repository.

A few things you could try:

- Push a small new commit to the repository to see if it triggers reindexing.
- Confirm that the repository isn't archived, empty, or exceeding any code search limitations.
- Check whether code search is enabled for your account and whether the issue occurs in both the new and legacy search experiences.
- Try searching from an incognito window or another browser to rule out a client-side issue.

Since you've been seeing the same message for an extended period, this may be an indexing problem on GitHub's side rather than anything you're doing wrong. If the issue persists, opening a GitHub Support ticket with the repository name and the exact error message would probably be the best next step.

Hopefully a member of the GitHub team can investigate the indexing status for `LeoDavidson/discourse-directoryopus`.
